components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java

```diff
@@ -238,7 +270,25 @@ public boolean checkPreference(String tenant, String connectorName, String prope
     public boolean checkPreference(String tenant, String connectorName, String propertyName, boolean defaultValue)
             throws PreferenceRetrievalClientException {
 
-        X509HostnameVerifier hv = new SelfRegistrationHostnameVerifier();
+        String propertyValue = getConnectorPropertyValue(tenant, connectorName, propertyName);
+        if (propertyValue != null) {
+            return Boolean.parseBoolean(propertyValue);
+        }
+        return defaultValue;
```

Comment:
why we are passing a default value as a param? can't we have a static instance for this if it's a default value?

Reply:
The default value here is a static value (true) and it is common to every property type. 
The value is set from here,

https://github.com/wso2/carbon-identity-framework/blob/12a151508ff2b70a174abc3973f63da851b39985/components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java#L257